ovos_core/skill_installer.py

```diff
@@ -0,0 +1,229 @@
+        if not self.config.get("allow_pip"):
+            LOG.error(InstallError.DISABLED.value)
+            self.play_error_sound()
+            self.bus.emit(message.reply("ovos.skills.install.failed",
+                                        {"error": InstallError.DISABLED.value}))
+            return
+
+        url = message.data["url"]
+        if self.validate_skill(url):
+            success = self.pip_install([f"git+{url}"])
+            if success:
+                self.bus.emit(message.reply("ovos.skills.install.complete"))
+            else:
+                self.bus.emit(message.reply("ovos.skills.install.failed",
+                                            {"error": InstallError.PIP_ERROR.value}))
+        else:
+            LOG.error("invalid skill url, does not appear to be a github skill")
+            self.play_error_sound()
+            self.bus.emit(message.reply("ovos.skills.install.failed",
+                                        {"error": InstallError.BAD_URL.value}))
+
+    def handle_uninstall_skill(self, message: Message):
+        if not self.config.get("allow_pip"):
+            LOG.error(InstallError.DISABLED.value)
+            self.play_error_sound()
+            self.bus.emit(message.reply("ovos.skills.uninstall.failed",
+                                        {"error": InstallError.DISABLED.value}))
+            return
+        # TODO
+        LOG.error("pip uninstall not yet implemented")
+        self.play_error_sound()
+        self.bus.emit(message.reply("ovos.skills.uninstall.failed",
+                                    {"error": "not implemented"}))
+
+    def handle_install_python(self, message: Message):
+        if not self.config.get("allow_pip"):
+            LOG.error(InstallError.DISABLED.value)
+            self.play_error_sound()
+            self.bus.emit(message.reply("ovos.pip.install.failed",
+                                        {"error": InstallError.DISABLED.value}))
+            return
+        pkgs = message.data["packages"]
+        if pkgs:
+            if self.pip_install(pkgs):
+                self.bus.emit(message.reply("ovos.pip.install.complete"))
+            else:
+                self.bus.emit(message.reply("ovos.pip.install.failed",
+                                            {"error": InstallError.PIP_ERROR.value}))
+        else:
+            self.bus.emit(message.reply("ovos.pip.install.failed",
+                                        {"error": InstallError.NO_PKGS.value}))
+
+    def handle_uninstall_python(self, message: Message):
+        if not self.config.get("allow_pip"):
+            LOG.error(InstallError.DISABLED.value)
+            self.play_error_sound()
+            self.bus.emit(message.reply("ovos.pip.uninstall.failed",
+                                        {"error": InstallError.DISABLED.value}))
+            return
+        pkgs = message.data["packages"]
+        if pkgs:
+            if self.pip_uninstall(pkgs):
```

Comment:
I'd include logs but I'm not seeing anything in ovos.log, or any of the other logs

Reply:
it's up to each distro to provide that file, but defaults for ovos-core are in the actual uninstall method! i don't think this one should be part of the bus message, if the bus message can provide a constraints file it can provide the right packages from the start too...

```
        if constraints:
            with open(constraints) as f:
                # remove version pinning and normalize _ to - (pip accepts both)
                cpkgs = [p.split("~")[0].split("<")[0].split(">")[0].split("=")[0].replace("_", "-")
                         for p in f.read().split("\n") if p.strip()]
        else:
            cpkgs = ["ovos-core", "ovos-utils", "ovos-plugin-manager",
                     "ovos-config", "ovos-bus-client", "ovos-workshop"]
```

Reply:
also in a vanilla mk1 the messages are ignored if you dont enable pip in mycroft.conf so the mk1 is safw ;)

Reply:
Apologies, I missed the `else` statement! That would cover the concern. Also, from my conf file:

```json
"skills": {
    "installer": {
         "allow_pip": true,
         "break_system_packages": false
    }
},
```

Reply:
Also, user error on the config, uninstall works 🎉 

```bash
ovos@mark1-dev:~ $ mana send-message-file ~/uninstall_pip.json
Message Sent
ovos@mark1-dev:~ $ tail -f ~/.local/state/mycroft/*.log  | grep pip
2023-09-19 12:39:55.754 - bus - ovos_messagebus.event_handler:on_message:37 - DEBUG - {"type": "ovos.pip.uninstall", "data": {"packages": ["spongebobify"]}, "context": {"client_name": "mana", "client": "mana", "source": ["mana"]}}
2023-09-19 12:39:55.773 - skills - ovos_core.skill_installer:pip_uninstall:135 - INFO - (pip) Uninstalling spongebobify
2023-09-19 12:39:55.777 - skills - ovos_core.skill_installer:pip_uninstall:137 - DEBUG - /usr/bin/python -m pip uninstall -y spongebobify
2023-09-19 12:40:00.330 - bus - ovos_messagebus.event_handler:on_message:37 - DEBUG - {"type": "ovos.pip.uninstall.complete", "data": {}, "context": {"client_name": "mana", "client": "mana", "source": ["mana"], "session": {"active_skills": [], "utterance_states": {}, "session_id": "c5c9d450-eb64-4d10-a17a-a5ad2c098795", "history": [[{"type": "gui.status.request", "data": {}, "context": {}}, 1695148783.1369834], [{"type": "gui.status.request", "data": {}, "context": {}}, 1695148788.2786684], [{"type": "gui.status.request", "data": {}, "context": {}}, 1695148793.4468806], [{"type": "gui.status.request", "data": {}, "context": {}}, 1695148798.9485786], [{"type": "mycroft.audio.play_sound", "data": {"uri": "snd/acknowledge.mp3"}, "context": {}}, 1695148800.323367], [{"type": "ovos.pip.uninstall.complete", "data": {}, "context": {}}, 1695148800.3241098]], "lang": "en-us", "valid_languages": ["en-us"], "context": {"timeout": 101708920560, "frame_stack": []}}}}
2023-09-19 12:40:04.059 - bus - ovos_messagebus.event_handler:on_message:37 - DEBUG - {"type": "gui.status.request", "data": {}, "context": {"session": {"active_skills": [], "utterance_states": {}, "session_id": "c5c9d450-eb64-4d10-a17a-a5ad2c098795", "history": [[{"type": "gui.status.request", "data": {}, "context": {}}, 1695148788.2786684], [{"type": "gui.status.request", "data": {}, "context": {}}, 1695148793.4468806], [{"type": "gui.status.request", "data": {}, "context": {}}, 1695148798.9485786], [{"type": "mycroft.audio.play_sound", "data": {"uri": "snd/acknowledge.mp3"}, "context": {}}, 1695148800.323367], [{"type": "ovos.pip.uninstall.complete", "data": {}, "context": {}}, 1695148800.3241098], [{"type": "gui.status.request", "data": {}, "context": {}}, 1695148804.056833]], "lang": "en-us", "valid_languages": ["en-us"], "context": {"timeout": 101708920560, "frame_stack": []}}}}
^C
ovos@mark1-dev:~ $ pip list | grep spongebobify
```